### PR TITLE
Migrate to Material3

### PR DESCRIPTION
**Pull request by migos**
* Migrate to Material3
* Update dependencies
* Remove unused koin's fragmentFactory

### CODE REVIEW

File: app/src/main/kotlin/com/tailoredapps/androidapptemplate/navigation/Navigation.kt [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
 import androidx.navigation.NavBackStackEntry
 import com.tailoredapps.androidapptemplate.R
 import com.tailoredapps.androidapptemplate.ui.overview.OverviewScreen
 
-sealed class Screen(
-    @StringRes val titleRes: Int,
-    val route: String,
-) {
+sealed class Screen {
+    abstract val titleRes: Int
```
Comment (FlowMo7) [inline app/src/main/kotlin/com/tailoredapps/androidapptemplate/navigation/Navigation.kt:9]: Is `@StringRes` not applicable to `abstract val`s, or is there another reason to not annotate this field hier? :thinking:

File: settings.gradle [hunk 2]
```diff
@@ -1,6 +1,10 @@
 plugins {
     // See https://jmfayard.github.io/refreshVersions
     id 'de.fayard.refreshVersions' version '0.40.2'
+////                           # available:'0.50.0'
+////                           # available:'0.50.1'
+////                           # available:'0.50.2'
+////                           # available:'0.51.0'
```
Comment (FlowMo7) [inline settings.gradle:7]: See above

File: app/src/main/kotlin/com/tailoredapps/androidapptemplate/navigation/Navigation.kt (hunk 1 above)
Comment (migos) [reply app/src/main/kotlin/com/tailoredapps/androidapptemplate/navigation/Navigation.kt:9]: sadly not applicable

File: base-ui/src/main/kotlin/com/tailoredapps/androidapptemplate/base/ui/theme/Color.kt [hunk 3]
```diff
@@ -2,7 +2,66 @@ package com.tailoredapps.androidapptemplate.base.ui.theme
...
+val md_theme_dark_surfaceVariant = Color(0xFF191C1E)
+val md_theme_dark_onSurfaceVariant = Color(0xFFC0C7CD)
+val md_theme_dark_outline = Color(0xFF8A9297)
+val md_theme_dark_inverseOnSurface = Color(0xFF191C1E)
+val md_theme_dark_inverseSurface = Color(0xFFE1E2E5)
+val md_theme_dark_inversePrimary = Color(0xFF006686)
+val md_theme_dark_shadow = Color(0xFF000000)
+val md_theme_dark_surfaceTint = Color(0xFF71D2FF)
+val md_theme_dark_outlineVariant = Color(0xFF40484C)
+val md_theme_dark_scrim = Color(0xFF000000)
```
[the remark this answers is not in the document]
Comment (migos) [reply base-ui/src/main/kotlin/com/tailoredapps/androidapptemplate/base/ui/theme/Color.kt:65]: copied from material tool builder 🤷

File: buildSrc/settings.gradle [hunk 4]
```diff
@@ -4,6 +4,10 @@ pluginManagement {
     }
     plugins {
         id 'de.fayard.refreshVersions' version '0.40.2'
+////                               # available:'0.50.0'
+////                               # available:'0.50.1'
+////                               # available:'0.50.2'
+////                               # available:'0.51.0'
```
[the remark this answers is not in the document]
Comment (migos) [reply buildSrc/settings.gradle:10]: haven't tried if this version is working again with buildSrc. would try this in another PR

**Review by FlowMo7 — APPROVED** [review]

**Pull request merged**